pkg/apis/migration/migration.go

```diff
@@ -162,6 +166,62 @@ func GetVolumeMigrationService(ctx context.Context, volumeManager *cnsvolume.Man
 			informer.Informer().Run(stopCh)
 			log.Debugf("Informer started for cnsvspherevolumemigrations")
 		}()
+		crdCleanupTicker := time.NewTicker(time.Duration(defaultCRDCleanupIntervalInMin * time.Minute))
+		defer crdCleanupTicker.Stop()
+		log.Infof("CRDs cleanup routine %s %+v", defaultCRDCleanupIntervalInMin, crdCleanupTicker)
+
+		//CRDs cleanup routine
+		go func() {
```

Comment:
Say the VCP PV had reclaimPolicy as `Retain`, and the devops deletes the PVC and PV(the CNS volume still resides in CNS). And the CR for this PV failed to be deleted during PV delete metadata sync logic. In this case I think the CR still remains as the current logic is comparing the CR with the CNS entry. Shouldn't you be comparing the CR with the VCP PV?

Reply:
Let's discuss this in PR: https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/440.